modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureRepository.java

```diff
@@ -87,6 +87,21 @@ public static final class Repository {
             DEFAULT_MAX_SINGLE_UPLOAD_SIZE,
             Property.NodeScope
         );
+
+        /**
+         * The batch size for batched delete requests
+         */
+        static final Setting<Integer> DELETION_BATCH_SIZE_SETTING = Setting.intSetting(
+            "delete_objects_max_size",
+            AzureBlobStore.MAX_ELEMENTS_PER_BATCH,
+            1,
+            AzureBlobStore.MAX_ELEMENTS_PER_BATCH
+        );
+
+        /**
+         * The maximum number of concurrent batch deletes
+         */
+        static final Setting<Integer> MAX_CONCURRENT_BATCH_DELETES_SETTING = Setting.intSetting("max_concurrent_batch_deletes", 10, 1);
```

Comment:
I suggest we give it a sensible max value, e.g. `100`.

Reply:
Added in 6c4697e0c61427c6ed79f25fc262e4bae91d597e